From 6da8ad295dcc9226bf716ac345d16fc66c06bc2c Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 17 May 2016 21:43:30 +0800
Subject: [PATCH 0528/2241] sdio: a3700: support SD card detection in DT node

In order to support SD card hot plug, interrupt based
SD card detection should be supported.
This patch add SD card detection support in DT node

Change-Id: I419c2cb1ccc2d89b7b8ffd2b5b88cccd8616e2b3
Signed-off-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29743
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
Verified-Armada37x0: Wilson Ding <dingwei@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 4 ++--
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   | 5 +++++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index ae07487..224c2b3 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -132,14 +132,14 @@
 			};
 
 			sdhci1: sdhci@d0000 {
-				broken-cd;
 				wp-inverted;
 				bus-width = <4>;
+				cd-gpios = <&gpio_sb 2 GPIO_ACTIVE_LOW>;
 				vqmmc-supply = <&vcc_sd_reg1>;
 				status = "okay";
 
 				pinctrl-names = "default";
-				pinctrl-0 = <&sdio_pins>;
+				pinctrl-0 = <&sdio_pins &cd_pins>;
 			};
 
 			sdhci0: sdhci@d8000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index fbe0b2b..026dd68 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -192,6 +192,11 @@
 					marvell,pins = "mpp3";
 					marvell,function = "gpio";
 				};
+
+				cd_pins: cd-pins {
+					marvell,pins = "mpp2";
+					marvell,function = "gpio";
+				};
 			};
 
 			cpufreq@14000 {
-- 
2.7.4

